test/integration/kubernetes.test.ts

```diff
@@ -65,6 +69,17 @@ tap.test('snyk-monitor container started', async (t) => {
   console.log('Done -- snyk-monitor exists!');
 });
 
+tap.test('create local container registry', async (t) => {
+  console.log('Creating local container registry...');
+  await exec('docker run -d --restart=always -p "5000:5000" --name "kind-registry" registry:2');
+  await exec('docker network connect "kind" "kind-registry"');
+  
+  console.log('Pushing python:rc-buster image to the local registry');
+  await kubectl.applyK8sYaml('./test/fixtures/insecure-registries/push-to-registry-job.yaml');
+
+  t.pass('successfully pushed image to a local registry');
```

Comment:
WDYT of this test being "create local container registry and push an image" and we could also await the K8s Job here, this way the test is self-contained. Otherwise the test later at line 244 does a wait on the Job but folks may wonder where this job is coming from?

Reply:
So I had it like that (contained in one test) but the job takes some sweet time, so moving the 'wait' to the actual test reduces how long the tests take, as by the time we get to the test that job might have even completed. I thought that was more efficient, what do you think?

Reply:
Hmm I see! We could put a big warning here that the job should be awaited as it takes some time, and rename this message as it is a bit misleading because the image is not yet pushed to the registry.

There is a way to speed up things: we could build our own Docker image with skopeo installed on it, let's say it's named `snyk/runtime-fixtures:skopeo-copy` (similarly to `snyk/runtime-fixtures:tinyproxy) and then use this image in the Job with the simple command of `skopeo copy from to`, this way the Job should be fairly quick! Just a thought really 😄 

My dislike for this is the disassociation between where we create the job and where we await them - 100s of lines later. For comparison the other tests are somewhat in order - first test creates 5-6 deployments, next test makes sure the deployments are sent to the upstream.

Reply:
Yeah, I totally agree, I really dislike the fact that setup is in 'test', because it's not a test and then that we are waiting for the job later on. 
I'll put a warning comment against it and I'm going to follow up with another PR with this optimisation to build our own image as it sounds good 👍 I'll add a note to the other task

Reply:
Thank you!

Reply:
Np! :-)